lib/TPP/Transforms/Vectorization.cpp

```diff
@@ -0,0 +1,109 @@
+//===- Vectorization.cpp -----------------------------------------*- C++-*-===//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements tile configuration hoisting on parallel loops.
+//
+//===----------------------------------------------------------------------===//
+#include "TPP/Dialect/Xsmm/XsmmUtils.h"
+#include "mlir/Dialect/Affine/Utils.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir {
+namespace tpp {
+#define GEN_PASS_DEF_VECTORIZATIONPASS
+#include "TPP/Passes.h.inc"
+} // namespace tpp
+} // namespace mlir
+
+using namespace mlir;
+using namespace std;
+
+namespace mlir {
+namespace tpp {
+
+struct LinalgGenericToVector : OpRewritePattern<linalg::GenericOp> {
+  using OpRewritePattern<linalg::GenericOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::GenericOp linalgOp,
+                                PatternRewriter &rewriter) const override {
+    if (xsmm::utils::getDataType(rewriter, linalgOp.getOperand(0).getType()) ==
```

Comment:
I take this is about VNNI layout?
It should be a separate pass as it is unrelated to vectorization itself.

Reply:
As discussed yesterday, this is specifcally for VNNI for vectorization, hence I am not moving this out.